clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/JsonHubProtocol.java

```diff
@@ -151,4 +141,42 @@ public TransferFormat getTransferFormat() {
     public String writeMessage(HubMessage hubMessage) {
         return gson.toJson(hubMessage) + RECORD_SEPARATOR;
     }
+
+    private ArrayList<Object> bindArguments(JsonArray argumentsToken, List<Class<?>> paramTypes) {
+        if (argumentsToken.size() != paramTypes.size()) {
+            throw new RuntimeException(String.format("Invocation provides %d argument(s) but target expects %d.", argumentsToken.size(), paramTypes.size()));
+        }
+
+        ArrayList<Object> arguments = null;
+        if (paramTypes != null && paramTypes.size() >= 1) {
+            arguments = new ArrayList<>();
+            for (int i = 0; i < paramTypes.size(); i++) {
+                arguments.add(gson.fromJson(argumentsToken.get(i), paramTypes.get(i)));
+            }
+        }
+
+        return arguments;
+    }
+
+    private ArrayList<Object> bindArguments(JsonReader reader, List<Class<?>> paramTypes) throws IOException {
+        reader.beginArray();
+        int paramCount = paramTypes.size();
+        int argCount = 0;
+        ArrayList<Object> arguments = new ArrayList<>();
+        while (reader.peek() != JsonToken.END_ARRAY) {
+            if (argCount < paramCount) {
+                arguments.add(gson.fromJson(reader, paramTypes.get(argCount)));
+            } else {
+                reader.skipValue();
+            }
+            ++argCount;
```

Comment:
nit: `argCount++`
The order should matter since you're not directly using whats returned here but I think we usually put the `++` after.